includes/class-wc-subscriptions-order.php

```diff
@@ -77,6 +77,22 @@ public static function init() {
 		add_filter( 'woocommerce_order_query_args', array( __CLASS__, 'map_order_query_args_for_subscriptions' ) );
 
 		add_filter( 'woocommerce_orders_table_query_clauses', [ __CLASS__, 'filter_orders_query_by_parent_orders' ], 10, 2 );
+
+		add_action( 'woocommerce_before_delete_order', [ __CLASS__, 'delete_order_update_order_related_subscriptions_last_order_date_created' ], 10, 2 );
```

Comment:
This is being triggered before the order is deleted so when `$subscription->get_time( 'last_order_date_created', 'gmt' );` gets called it loads the date of the order being deleted. We should use `woocommerce_trash_order` instead